pkg/handlers/metrics.go

```diff
@@ -0,0 +1,86 @@
+package handlers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/facebookgo/clock"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+type fetchMetrics func(context context.Context, startTime time.Time, endTime time.Time) (models.MetricsDigest, error)
+
+// MetricsHandler is the handler for retrieving metrics
+type MetricsHandler struct {
+	FetchMetrics fetchMetrics
+	Logger       *zap.Logger
+	Clock        clock.Clock
+}
+
+// Handle handles a web request and returns a metrics digest
+func (h MetricsHandler) Handle() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		logger, ok := appcontext.Logger(r.Context())
+		if !ok {
+			h.Logger.Error("Failed to logger from context in metrics handler")
+			logger = h.Logger
+		}
+		switch r.Method {
+		case "GET":
+			// get our time params
+			startTimeParam, ok := r.URL.Query()["startTime"]
+			if !ok || len(startTimeParam[0]) < 1 {
```

Comment:
Right now, this is the only endpoint with time involved, but if that changes in the future, we'll probably want to move this work into a shared package for param parsing.